src/Microsoft.Azure.SignalR.Common/Utilities/AuthenticationHelper.cs

```diff
@@ -34,13 +37,20 @@ public static string GenerateAccessToken(string signingKey, string audience, IEn
         {
             var expire = DateTime.UtcNow.Add(lifetime);
 
-            return GenerateJwtBearer(
+            var jwtToken = GenerateJwtBearer(
                 audience: audience,
                 claims: claims,
                 expires: expire,
                 signingKey: signingKey,
                 requestId: requestId
             );
+
+            if (jwtToken.Length > MaxTokenLength)
+            {
+                throw new ArgumentException("AccessToken must not be longer than 4K");
```

Comment:
For server connections, if the token too long, we should make sure exception throws.

For client’ negotiation, consider return “413 Request Entity Too Large” or “414 Request-URI Too Long”

Reply:
Right. Update to return 413 for client which is more accurate. And tested locally with expected behavior.